CONTRIBUTING.md

```diff
@@ -0,0 +1,320 @@
+    - `nosetests` finds all the appropriate tests and runs them
+
+### Running tests via Docker
+
+The basics should work with basically no further setup. In the terminal, `cd` to the directory where you cloned dbt. So, for example, if you cloned dbt to `~/git/dbt`:
+
+```
+cd ~/git/dbt
+```
+
+Then you'll want to make a test.env file. Fortunately, there's a sample which is fine for our purposes:
+
+```
+cp test.env.sample test.env
+```
+
+If you want to test snowflake/bigquery/redshift locally you'll need to get credentials and add them to this file. But, to start, you can just focus on postgres tests. They have the best coverage, are the fastest, and are the easiest to set up.
+
+To run the unit tests, use `make test-unit` - it will run the unit tests on python 2.7 and 3.6, and a pep8 linter.
+
+To run the postgres+python 3.6 integration tests, you'll have to do one extra step of setting up the database:
+
+```
+docker-compose up -d database
+PGHOST=localhost PGUSER=root PGPASSWORD=password PGDATABASE=postgres bash test/setup_db.sh
+```
+
+And then to actually run them, you can do `make test-quick`.
+
+If you want to see what exactly is getting run on these commands, look at the `Makefile`. Note that the commands start with an `@` which you can ignore, just makefile magic. If you want to see what the involved `tox` commands are using, look at the corresponding `tox.ini` section - hopefully it's pretty self-explanatory.
+
+### Running tests in CI
+
+When a contributor to dbt pushes code, GitHub will trigger a series of CI builds on CircleCI and Appveyor (Windows) to test all of dbt's code. The CI builds trigger all the integration tests, not just postgres+python3.6.
+
+The Snowflake tests take a very long time to run (about an hour), so don't just sit around waiting, it'll be a while!
+
+If you open a PR as a non-contributor, these tests won't run automatically. Someone from the dbt team will reach out to you and get them running after reviewing your code.
+
+## Running dbt locally
+
+Sometimes, you're going to have to pretend to be an end user to reproduce bugs and stuff. So that means manually setting up some stuff that the test harness takes care of for you.
+
+### installation
+
+First, from the `dbt` directory, install dbt in 'editable' mode. There are a couple ways to do it, but I'm in the habit of `pip install -e .`, which tells pip to install the package in the current directory in "editable" mode. What's cool about this mode is any changes you make to the current dbt directory will be reflected immediately in your next `dbt` run.
+
+### Profile
+
+Now you'll also need a 'dbt profile' so dbt can tell how to connect to your database. By default, this file belongs at `~/.dbt/profiles.yml`, so `mkdir ~/.dbt` and then open your favorite text editor and write out something like this to `~/.dbt/profiles.yml`:
+
+```
+config:
+    send_anonymous_usage_stats: True
```

Comment:
Drew had some feedback on my initial write-up that we should encourage devs to set this to `False`, which seemed reasonable